### PR TITLE
feat: copied OpenAPI document from rivebank repo

### PR DESCRIPTION
**Pull request by amsteel**
Copied OpenAPI document from [rivebank](https://github.com/delta-incubator/riverbank/blob/main/api-description.ymll)

### CODE REVIEW

File: delta-sharing-protocl-api-description.yml [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+
+
+  '/shares/{share}/schemas/{schema}/tables/{table}/query':
+    post:
+      operationId: 'QueryTable'
+      summary: 'Query the table'
+      tags:
+        - tables
+        - official
+      parameters: *tableParameters
```
Comment (linzhou-db) [inline delta-sharing-protocl-api-description.yml:324]: We need a `queyTableParameters` and add all the supported parameters: https://github.com/delta-io/delta-sharing/blob/main/PROTOCOL.md#request-body

File: delta-sharing-protocl-api-description.yml [hunk 2]
```diff
@@ -0,0 +1,512 @@
...
+              description: 'A long value which represents the current table version'
+              schema:
+                type: integer
+          content:
+            'application/x-ndjson':
+              schema:
+                $ref: '#/components/schemas/TableMetadataResponse'
+
+
+  '/shares/{share}/schemas/{schema}/tables/{table}/query':
```
Comment (linzhou-db) [inline delta-sharing-protocl-api-description.yml:317]: '/shares/{share}/schemas/{schema}/tables/{table}/changes' is missing from the doc

**Review by linzhou-db — COMMENTED** [review]
Sorry for the delay, took a initial pass.

**Comment by linzhou-db** [comment]
@amsteel did a initial review

File: delta-sharing-protocl-api-description.yml [hunk 3]
```diff
@@ -0,0 +1,512 @@
...
+    The Delta Sharing protocol is still very early in its development.
+  version: "0.2"
+  title: Delta Sharing Protocol
+  contact:
+    email: ""
+  license:
+    name: "AGPL v3.0"
+    url: "https://www.gnu.org/licenses/agpl-3.0.en.html"
+
+servers:
```
[the remark this answers is not in the document]
Comment (amsteel) [reply delta-sharing-protocl-api-description.yml:14]: from the riverbank project. Please let me know the correct url to be used.

[the remark this answers is not in the document]
Comment (amsteel) [reply delta-sharing-protocl-api-description.yml:14]: from the riverbank project. Please let me know the correct url to be used

File: delta-sharing-protocl-api-description.yml [hunk 4]
```diff
@@ -0,0 +1,512 @@
...
+          $ref: "#/components/responses/401"
+        '403':
+          $ref: "#/components/responses/403"
+        '404':
+          $ref: "#/components/responses/404"
+        '500':
+          $ref: "#/components/responses/500"
+        '200':
+          description: 'The share''s metadata was successfully returned'
+          headers:
```
[the remark this answers is not in the document]
Comment (amsteel) [reply delta-sharing-protocl-api-description.yml:112]: Updated

File: delta-sharing-protocl-api-description.yml [hunk 5]
```diff
@@ -0,0 +1,512 @@
...
+        '200':
+          description: |
+            The user could successfully list the tables for the given schema in the given share
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ListTablesResponse'
+
+  '/shares/{share}/schemas/{schema}/tables/{table}':
+    head:
```
[the remark this answers is not in the document]
Comment (amsteel) [reply delta-sharing-protocl-api-description.yml:232]: Updated

File: delta-sharing-protocl-api-description.yml (hunk 1 above)
Comment (amsteel) [reply delta-sharing-protocl-api-description.yml:324]: It is part of #/components/schemas/QueryRequest, updated with the md content.

File: delta-sharing-protocl-api-description.yml (hunk 2 above)
Comment (amsteel) [reply delta-sharing-protocl-api-description.yml:317]: added

**Comment by amsteel** [comment]
@linzhou-db updated the doc

File: delta-sharing-protocl-api-description.yml [hunk 6]
```diff
@@ -0,0 +1,692 @@
...
+
+
+paths:
+  '/shares':
+    get:
+      operationId: 'ListShares'
+      summary: 'List the shares accessible to the recipient'
+      description: |
+        Without any query parameters the request will return the first page
+        of "shares" available to the authenticated recipent
```
[the remark this answers is not in the document]
Comment (amsteel) [reply delta-sharing-protocl-api-description.yml:42]: I don't think there is one defined as the md document. I think it is better to have one defined.

[the remark this answers is not in the document]
Comment (amsteel) [reply delta-sharing-protocl-api-description.yml:42]: @linzhou-db Comments? 
```suggestion
        Default page size as 100
```

File: delta-sharing-protocl-api-description.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (linzhou-db) [reply delta-sharing-protocl-api-description.yml:14]: I actually need to understand what's the server for? server of what?

File: delta-sharing-protocl-api-description.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (linzhou-db) [reply delta-sharing-protocl-api-description.yml]: it is set to 500 as default  in the server.

File: delta-sharing-protocl-api-description.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (amsteel) [reply delta-sharing-protocl-api-description.yml:14]: quote from openapi document: In OpenAPI 3.0, you use the servers array to specify one or more base URLs for your API. servers replaces the host, basePath and schemes keywords used in OpenAPI 2.0. Each server has an url and an optional Markdown-formatted description.

File: delta-sharing-protocl-api-description.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (amsteel) [reply delta-sharing-protocl-api-description.yml]: Updated to 500

File: delta-sharing-protocl-api-description.yml [hunk 7]
```diff
@@ -0,0 +1,695 @@
...
+          $ref: "#/components/responses/500"
+        '200':
+          description: |
+            The user could successfully list the tables for the given schema in the given share
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ListTablesResponse'
+
+  '/shares/{share}/schemas/{schema}/tables/{table}':
```
Comment (linzhou-db) [inline delta-sharing-protocl-api-description.yml:229]: /shares/{share}/schemas/{schema}/tables/{table}/version

File: delta-sharing-protocl-api-description.yml [hunk 8]
```diff
@@ -0,0 +1,695 @@
...
+    get:
+      operationId: 'GetTableVersion'
+      summary: 'Return the latest version identifier for the named table'
```
Comment (linzhou-db) [inline delta-sharing-protocl-api-description.yml:232]: 'Return the table version'

File: delta-sharing-protocl-api-description.yml [hunk 9]
```diff
@@ -0,0 +1,695 @@
...
+          $ref: "#/components/responses/500"
+        '200':
+          description: |
+            The user could successfully list the schemas in the given share
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ListSchemasResponse'
+
+  '/shares/{share}/schemas/{schema}/tables':
```
Comment (linzhou-db) [inline delta-sharing-protocl-api-description.yml:170]: 'shares/{share}/all-tables' is missing from the doc.

**Review by linzhou-db — COMMENTED** [review]
almost there.

File: delta-sharing-protocl-api-description.yml (hunk 9 above)
Comment (amsteel) [reply delta-sharing-protocl-api-description.yml:170]: added

File: delta-sharing-protocl-api-description.yml (hunk 8 above)
Comment (amsteel) [reply delta-sharing-protocl-api-description.yml:232]: updated

File: delta-sharing-protocl-api-description.yml [hunk 10]
```diff
@@ -0,0 +1,744 @@
...
+          description: |
+            The user could successfully list the tables for the given schema in the given share
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ListTablesResponse'
+
+  '/shares/{share}/all-tables':
+    get:
+      operationId: 'ListTables'
```
Comment (linzhou-db) [inline delta-sharing-protocl-api-description.yml:228]: nit: ListAllTables

**Review by linzhou-db — APPROVED** [review]
only one last change

**Pull request merged**